File: source/js/squan-logic/layer.js
// This will be where the logic for each virtual Square-1 layer will be

export class Layer { 
    constructor(type) {
        this.type = type;

        const pieces = [2, 1, 2, 1, 2, 1, 2, 1];
    }
    turnClockwise() {
        if (this.type == "top") {
            let lastPiece = pieces.shift();
            pieces.push(lastPiece)
            // idk how js works
        }
    }
    turnCounterclockwise() {
        console.log("U'/D'");
    }
}

File: source/js/squan-logic/layer.py
""" Logic for the top and bottom layers of the virtual Square-1 """


class Layer:
    """Single layer object"""

    def __init__(self, top_or_bottom, pieces):
        self.top_or_bottom = top_or_bottom
        self.pieces = pieces

    def tick_clock(self):
        """Turn the virtual Square-1 one piece clockwise"""

    def tick_prime(self):
        """Turn the virtual Square-1 one piece counterclockwise"""
